torch/csrc/api/include/torch/optim/optimizer.h

```diff
@@ -26,32 +30,117 @@ class InputArchive;
 
 namespace torch {
 namespace optim {
+
+class TORCH_API OptimizerParamState {
+ public:
+  virtual std::unique_ptr<OptimizerParamState> clone() const;
+  virtual void serialize(torch::serialize::InputArchive& archive);
+  virtual void serialize(torch::serialize::OutputArchive& archive) const;
+  virtual ~OptimizerParamState() = default;
+};
+
+template <typename Derived>
+class TORCH_API OptimizerCloneableParamState : public OptimizerParamState {
+  std::unique_ptr<OptimizerParamState> clone() const override {
+    return std::make_unique<Derived>(static_cast<const Derived&>(*this));
+  }
+};
+
+class TORCH_API OptimizerOptions {
+ public:
+  virtual std::unique_ptr<OptimizerOptions> clone() const;
+  virtual void serialize(torch::serialize::InputArchive& archive);
```

Comment:
Could this be `const torch::serialize::InputArchive& archive`?

Reply:
I didn't do it because none of the existing functions in serialize.h had it const, and I am not entirely sure if we can

Reply:
ah I see, let's keep it the current way then :)